tests/semanticdb/expect/dep-match.scala

```diff
@@ -0,0 +1,9 @@
+object Test_depmatch {
+  type Foo = Int { type U }
+  type Bar[T] = T match {
+    case Unit => Unit
+  }
+  inline def baz(foo: Foo): Unit = {
+    val v: Bar[foo.U] = ???
```

Comment:
Another case where `<pre of foo.U>.widen` returns `TypeRef` to `Foo`, and requires `dealias` to retrieve the refinedtype